mach/proto/as/comm4.c

```diff
@@ -243,8 +244,8 @@ archive(void) {
 		if (needed()) {
 			fseek(input,offset,0);
 			archsize = header.ar_size;
-			header.ar_name[14] = '\0';
-			parse(remember(header.ar_name));
+			ar_name[AR_NAME_MAX] = '\0';
```

Comment:
Buffer overflow; this should be AR_NAME_MAX-1 (or ar_hdr needs to define the field as char[AR_NAME_MAX+1]. Looks like this bug has always been here. I've seen similar bugs before.